backend/experiment/rules/toontjehoger_3_plink.py

```diff
@@ -24,6 +25,41 @@ class ToontjeHoger3Plink(Base):
     SCORE_EXTRA_2_CORRECT = 4
     SCORE_EXTRA_WRONG = 0
 
+    def validate_playlist(self, playlist: Playlist):
+        """ The original Toontjehoger (Plink) playlist has the following format:
+        ```
+        Billy Joel,Piano Man,0.0,1.0,toontjehoger/plink/2021-005.mp3,70s,vrolijk
+        Boudewijn de Groot,Avond,0.0,1.0,toontjehoger/plink/2021-010.mp3,90s,tederheid
+        Bruce Springsteen,The River,0.0,1.0,toontjehoger/plink/2021-016.mp3,80s,droevig
+        ```
+        """
+        errors = []
+        sections = playlist.section_set.all()
+        if not [s.song for s in sections]:
+            errors.append(
+                'Sections should have associated song objects.')
+        artist_titles = sections.values_list(
+            'song__name', 'song__artist').distinct()
+        if len(artist_titles) != len(sections):
+            errors.append(
+                'Sections should have unique combinations of song.artist and song.name fields.')
+        self.validate_era_and_mood(sections, errors)
+        return errors
+
+    def validate_era_and_mood(self, sections, errors):
+        eras = sorted(sections.order_by('tag').values_list(
+            'tag', flat=True).distinct())
+        if not all(re.match(r'[0-9]0s', e) for e in eras):
+            errors.append(
+                'The sections should be tagged with an era in the format [0-9]0s, e.g., 90s')
+        moods = sorted(sections.order_by('group').values_list(
+            'group', flat=True).distinct())
+        if 'droevig' not in moods:
+            errors.append(
+                "The sections' groups should be indications of the songs' moods in Dutch"
+            )
+        return errors
```

Comment:
Does this mean only "droevig" is a required mood?

Reply:
Yeah, it's difficult to foresee the exact labels. I don't have a better idea at the moment.